### PR TITLE
embrace zstm

### PR DESCRIPTION
**Pull request by mschuwalow**
No logic changes, just cleanup

### CODE REVIEW

File: membership/src/main/scala/zio/membership/hyparview/package.scala [hunk 1]
```diff
@@ -132,75 +132,68 @@ package object hyparview {
...
-                                                ActiveProtocol
-                                                  .ForwardJoin(views.myself, msg.sender, TimeToLive(config.arwl))
-                                              )
-                                        )
-                                  reply <- ByteCodec
-                                            .toChunk(JoinReply(views.myself))
-                                            .mapError(e => zio.membership.SerializationError(e.msg))
-                                  _ <- con.send(reply)
-                                } yield Some(msg.sender)
+                                  full <- Views.isActiveViewFull
```
Comment (mijicd) [inline membership/src/main/scala/zio/membership/hyparview/package.scala:156]: We have `ifM` as well :).

File: membership/src/main/scala/zio/membership/hyparview/package.scala [hunk 2]
```diff
@@ -228,17 +221,14 @@ package object hyparview {
...
-                    activeViewSize <- views.activeViewSize
-                    _              <- STM.check(activeViewSize < views.activeViewCapacity)
-                    nodeOpt        <- views.passiveView.flatMap(xs => tRandom.selectOne(xs.toList))
-                    node           <- nodeOpt.fold[STM[Nothing, T]](STM.retry)(STM.succeed(_))
-                  } yield (node, Neighbor(views.myself, activeViewSize <= 0))
-                }
-              }
+            ZSTM.atomically {
+              for {
+                _              <- Views.isActiveViewFull.flatMap(full => STM.check(!full))
```
Comment (mijicd) [inline membership/src/main/scala/zio/membership/hyparview/package.scala:226]: You can use `retryWhile` or `retryUntil` here.

File: membership/src/main/scala/zio/membership/hyparview/periodic.scala [hunk 3]
```diff
@@ -9,48 +9,43 @@ import zio.logging.Logging.Logging
...
+                    node,
+                    ActiveProtocol
+                      .Shuffle(localAddr, localAddr, active, passive, TimeToLive(config.shuffleTTL))
+                  )
+                  .as(state)
+            }
+          }
+          .commit
+          .flatten
+      go.eventually
```
Comment (mijicd) [inline membership/src/main/scala/zio/membership/hyparview/periodic.scala:36]: You can remove val here :).

File: membership/src/main/scala/zio/membership/hyparview/periodic.scala [hunk 4]
```diff
@@ -9,48 +9,43 @@ import zio.logging.Logging.Logging
...
-          go.eventually
-        }
-      }
+    getConfig.flatMap { config =>
+      val go: ZIO[TRandom with Views[T], SendError, ViewState] =
+        Views.activeView
+          .map(_.toList)
+          .flatMap { nodes =>
+            TRandom.selectOne(nodes).flatMap {
+              case None => STM.succeed(Views.viewState.commit)
```
Comment (mijicd) [inline membership/src/main/scala/zio/membership/hyparview/periodic.scala:18]: Why not `Views.viewState` only?

Comment (mschuwalow) [reply membership/src/main/scala/zio/membership/hyparview/periodic.scala:18]: It expects an STM[_, UIO[_]], so I rolled with this one

**Comment by mschuwalow** [comment]
@mijicd thanks for your review. I think I addressed most of the points, can you take another look?

I'm also running into flakiness with the tests as they are currently using a fixed port for the TCPTransportSpec. Sadly I can't think of a good way to fix that without polluting our interfaces. Maybe you can take a look?

**Pull request merged**